CMakeLists.txt

```diff
@@ -17,7 +17,7 @@ option(FAASM_SGX_SUPPORT "Enable/Disable SGX Support" OFF)
 option(FAASM_SGX_SIM_MODE "SGX sim mode flag" ON)
 option(FAASM_SGX_ATTESTATION "Add support for attestation" OFF)
 option(FAASM_SGX_XRA "Add support for XRA" OFF)
-option(FAASM_SGX_WAMR_AOT_MODE "Turns AoT on / off (vs. Interpreter) within the enclave" ON)
+option(FAASM_SGX_WAMR_AOT_MODE "Turns AoT on / off (vs. Interpreter) within the enclave (Attention: AoT needs SGXv2!)" ON)
```

Comment:
Is there any way we can auto-detect the SGX version and fail if they have this switched on and not the right version of SGX? We could perhaps modify our `cmake/FindSGX.cmake` file (https://github.com/faasm/faasm/blob/sgx-tests/cmake/FindSGX.cmake, links to the PR so link will break once it's merged), to output a variable like `IS_SGX_V2`, then have:

```
if(FAASM_SGX_WAMR_AOT_MODE AND !IS_SGX_V2)
    # Print some error message and fail
endif()
```

Reply:
yes, auto-detection of SGX version should work quite easy. I'll look into it